static/app/views/settings/account/accountSecurity/accountSecurityWrapper.tsx

```diff
@@ -1,123 +1,124 @@
 import {cloneElement} from 'react';
-import type {RouteComponentProps} from 'react-router';
 
 import {addErrorMessage} from 'sentry/actionCreators/indicator';
 import {fetchOrganizations} from 'sentry/actionCreators/organizations';
-import DeprecatedAsyncComponent from 'sentry/components/deprecatedAsyncComponent';
+import LoadingError from 'sentry/components/loadingError';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
 import {t} from 'sentry/locale';
 import type {Authenticator} from 'sentry/types/auth';
 import type {OrganizationSummary} from 'sentry/types/organization';
 import type {UserEmail} from 'sentry/types/user';
 import {defined} from 'sentry/utils';
+import {useApiQuery, useMutation, useQuery} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import {useParams} from 'sentry/utils/useParams';
 
 const ENDPOINT = '/users/me/authenticators/';
 
-type Props = {
+interface Props {
   children: React.ReactElement;
-} & RouteComponentProps<{authId: string}, {}> &
-  DeprecatedAsyncComponent['props'];
-
-type State = {
-  emails: UserEmail[];
-  loadingOrganizations: boolean;
-  authenticators?: Authenticator[] | null;
-  organizations?: OrganizationSummary[];
-} & DeprecatedAsyncComponent['state'];
-
-class AccountSecurityWrapper extends DeprecatedAsyncComponent<Props, State> {
-  async fetchOrganizations() {
-    try {
-      this.setState({loadingOrganizations: true});
-      const organizations = await fetchOrganizations(this.api);
-      this.setState({organizations, loadingOrganizations: false});
-    } catch (e) {
-      this.setState({error: true, loadingOrganizations: false});
-    }
-  }
-
-  get shouldRenderLoading() {
-    return super.shouldRenderLoading || this.state.loadingOrganizations === true;
-  }
-
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    return [
-      ['authenticators', ENDPOINT],
-      ['emails', '/users/me/emails/'],
-    ];
-  }
-
-  componentDidMount() {
-    super.componentDidMount();
-    this.fetchOrganizations();
-  }
-
-  reloadData() {
-    this.fetchOrganizations();
-    super.reloadData();
-  }
-
-  handleDisable = async (auth: Authenticator) => {
-    if (!auth || !auth.authId) {
-      return;
-    }
+}
 
-    this.setState({loading: true});
+function AccountSecurityWrapper({children}: Props) {
+  const api = useApi();
+  const {authId} = useParams<{authId?: string}>();
+
+  const orgRequest = useQuery<OrganizationSummary[]>(
+    ['organizations'],
+    () => fetchOrganizations(api),
+    {staleTime: 0}
+  );
+  const emailsRequest = useApiQuery<UserEmail[]>(['/users/me/emails/'], {staleTime: 0});
+  const authenticatorsRequest = useApiQuery<Authenticator[]>([ENDPOINT], {staleTime: 0});
+
+  const handleRefresh = () => {
```

Comment:
we are using `useCallback` everywhere for handlers...shouldn't we use it here too?